@nrwl-angular/storybook-configuration miss to install @storybook/core-common

### Current Behavior

I added storybook to one Angular library and it installed the dependencies and configuration files, but when I tried to run the storybook task it fails to compile due to `@storybook/core-common` is missing. In order to fix it I just install `@storybook/core-common`

The plugin install the following dependencies
- @nrwl/storybook
-  @storybook/addon-essentials
-  @storybook/angular
-  @storybook/builder-webpack5
-  @storybook/core-server
-  @storybook/manager-webpack5
-  @storybook/test-runner
-  html-webpack-plugin
-  webpack

### Expected Behavior

To also install `@storybook/core-common` as a devDependency

### Github Repo

https://github.com/leosuncin/nrwl-nx-14259

### Steps to Reproduce

1. Create a workspace with Angular
2. Create an Angular library
3. Add a component to the library and register it in the module
4. Run @nrwl-angular/storybook-configuration with tsConfiguration enable: ``nx generate @nrwl/angular:storybook-configuration library-name --no-configureCypress --configureTestRunner --no-generateCypressSpecs --tsConfiguration --no-interactive``
5. Run the storybook for the project ``nx run library-name:storybook``
6. The storybook won't compile due to `@storybook/core-common` is not installed <img width="1110" alt="Captura de pantalla 2023-01-10 a la(s) 13 30 42" src="https://user-images.githubusercontent.com/4307697/211644141-6d7712b9-96fa-4f2e-ae78-3a6cdb3a7434.png">


### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 19.3.0
   OS   : darwin x64
   pnpm : 7.22.0
   
   nx : 15.4.5
   @nrwl/angular : 15.4.5
   @nrwl/cypress : 15.4.5
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.4.5
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.5
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.5
   @nrwl/js : 15.4.5
   @nrwl/linter : 15.4.5
   @nrwl/nest : 15.4.5
   @nrwl/next : Not Found
   @nrwl/node : 15.4.5
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.4.5
   @nrwl/web : Not Found
   @nrwl/webpack : 15.4.5
   @nrwl/workspace : 15.4.5
   @nrwl/vite : Not Found
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
   	 @storybook/angular: 6.5.15
```


### Failure Logs

```shell
ERR! TSError: ⨯ Unable to compile TypeScript:
ERR! libs/contact-list-module/.storybook/main.ts:3:47 - error TS2307: Cannot find module '@storybook/core-common' or its corresponding type declarations.
ERR! 
ERR! 3 import type { StorybookConfig, Options } from '@storybook/core-common';
ERR!                                                 ~~~~~~~~~~~~~~~~~~~~~~~~
ERR! libs/contact-list-module/.storybook/main.ts:14:24 - error TS7006: Parameter 'config' implicitly has an 'any' type.
ERR! 
ERR! 14   webpackFinal: async (config, { configType }: Options) => {
ERR!                           ~~~~~~
ERR! 
ERR!     at createTSError (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:859:12)
ERR!     at reportTSError (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:863:19)
ERR!     at getOutput (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:1077:36)
ERR!     at Object.compile (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:1433:41)
ERR!     at Module.m._compile (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:1617:30)
ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
ERR!     at Object.require.extensions.<computed> [as .ts] (/Users/user/nx-workspace/node_modules/.pnpm/ts-node@10.9.1_rhpwi32gjip3gggyfzgffz3q2q/node_modules/ts-node/src/index.ts:1621:12)
ERR!     at Module.load (node:internal/modules/cjs/loader:1081:32)
ERR!     at Function.Module._load (node:internal/modules/cjs/loader:922:12)
ERR!     at Module.require (node:internal/modules/cjs/loader:1105:19)
ERR!  libs/contact-list-module/.storybook/main.ts:3:47 - error TS2307: Cannot find module '@storybook/core-common' or its corresponding type declarations.
ERR! 
ERR! 3 import type { StorybookConfig, Options } from '@storybook/core-common';
ERR!                                                 ~~~~~~~~~~~~~~~~~~~~~~~~
ERR! libs/contact-list-module/.storybook/main.ts:14:24 - error TS7006: Parameter 'config' implicitly has an 'any' type.
ERR! 
ERR! 14   webpackFinal: async (config, { configType }: Options) => {
```


### Additional Information

EDIT: added a reproduction repository

--- a/packages/storybook/src/generators/configuration/configuration.ts
+++ b/packages/storybook/src/generators/configuration/configuration.ts
@@ -32,6 +32,7 @@ import {
   storybookNextAddonVersion,
   storybookSwcAddonVersion,
   storybookTestRunnerVersion,
+  storybookVersion,
 } from '../../utils/versions';
 
 export async function configurationGenerator(
@@ -120,6 +121,18 @@ export async function configurationGenerator(
   } else {
     logger.warn(
       `There is already an e2e project setup for ${schema.name}, called ${e2eProject}.`
+    );
+  }
+
+  if (schema.tsConfiguration) {
+    tasks.push(
+      addDependenciesToPackageJson(
+        tree,
+        {},
+        {
+          ['@storybook/core-common']: storybookVersion,
+        }
+      )
     );
   }
 

